nautobot/extras/tests/test_dynamicgroups.py

```diff
@@ -346,6 +347,50 @@ def test_map_filter_fields_skip_missing(self):
         finally:
             del group.model.dynamic_group_skip_missing_fields
 
+    def test_map_filter_fields_skip_method_filters_generate_query(self):
+        """
+        Test that method filters are skipped in `DynamicGroup._map_filter_fields` when the filterset
+        for the group's content type has a method named `generate_query_{filter_method}`.
+        """
+        group = self.groups[0]
+        filterset = group.filterset_class()
+        fields = group._map_filter_fields
+
+        # We know that `has_primary_ip` fits this bill, so let's test that.
+        field_name = "has_primary_ip"
+        filter_field = filterset.filters[field_name]
+
+        # Make some field presence assertions
+        self.assertIn(field_name, fields)
+
+        # The filterset should have the method name and `generate_query_` method
+        self.assertTrue(hasattr(filterset, filter_field.method))
+        self.assertTrue(hasattr(filterset, "generate_query_" + filter_field.method))
```

Comment:
Where are we testing the query construction that involves the `generate_query_*` methods?

Reply:
We aren't since it's going to vary wildly by method. Are you thinking we should test at least one that we know works a certain way to assert it?

Reply:
Yeah, testing that the dynamic group correctly constructs the appropriate query for a given set of filters that rely on this feature.